pkg/backends/fake/fake.go

```diff
@@ -0,0 +1,117 @@
+package fake
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+	batchV1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	loadTestV1 "github.com/hellofresh/kangal/pkg/kubernetes/apis/loadtest/v1"
+)
+
+var (
+	sleepImage = "alpine"
+	imageTag   = "latest"
+)
+
+// Fake enables the controller to run a LoadTest using Fake load provider which simulates load test
+type Fake struct {
+	kubeClient kubernetes.Interface
+	loadTest   *loadTestV1.LoadTest
+	logger     *zap.Logger
+}
+
+//New initializes new Fake provider handler to manage load test resources with Kangal Controller
+func New(kubeClientSet kubernetes.Interface, lt *loadTestV1.LoadTest, logger *zap.Logger) *Fake {
+	return &Fake{
+		kubeClient: kubeClientSet,
+		loadTest:   lt,
+		logger:     logger,
+	}
+}
+
+// SetDefaults set default values for creating a Fake LoadTest pods
+func (c *Fake) SetDefaults() error {
+	if c.loadTest.Status.Phase == "" {
+		c.loadTest.Status.Phase = loadTestV1.LoadTestCreating
+	}
+
+	if c.loadTest.Spec.MasterConfig.Image == "" {
+		c.loadTest.Spec.MasterConfig.Image = sleepImage
+	}
+
+	if c.loadTest.Spec.MasterConfig.Tag == "" {
+		c.loadTest.Spec.MasterConfig.Tag = imageTag
+	}
+
+	return nil
+}
+
+// CheckOrCreateResources check if Fake kubernetes resources have been create, if they have not been create them
+func (c *Fake) CheckOrCreateResources(ctx context.Context) error {
+	// Get the Namespace resource
+	namespace, err := c.kubeClient.CoreV1().Namespaces().Get(ctx, c.loadTest.Status.Namespace, metaV1.GetOptions{})
+	if err != nil {
+		// The LoadTest resource may no longer exist, in which case we stop
+		// processing.
+		if errors.IsNotFound(err) {
```

Comment:
is it fine to ignore all other possible errors here?

Reply:
Good point, that was like that before but I changed it to return any possible errors to the controller - so it will reschedule the event.